### PR TITLE
[#LIEF-9737] github-jira-bot: Parse headings

### PR DESCRIPTION
**Pull request by jonhue**

### CODE REVIEW

**Comment by jonhue** [comment]
QA:

* test that headings in Jira descriptions are being rendered correctly on GitHub
* test that headings in GitHub comments are being rendered correctly in Jira

File: lib/parser/heading.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Parser
+  class Heading
```
Comment (adamniedzielski) [inline lib/parser/heading.rb]: I think that it makes more sense to have `Parser::GithubToJira::Heading`, `Parser::JiraToGithub::Heading` than these two methods in a single class. We don't share code between the implementations and at the point of calling the code we always know in which direction we want to perform the translation.

Comment (jonhue) [reply lib/parser/heading.rb]: Agree 💯

**Comment by adamniedzielski** [comment]
👍

**Comment by jonhue** [comment]
QA:

1. when opening a pull request in the `github-jira-bot-test` repo with a corresponding Jira ticket, the headings in the Jira ticket description should be displayed correctly in the GitHub comment created by the bot.
2. when creating a `QA:` comment in a pull request in the `github-jira-bot-test` repo which contains headings, these headings should be displayed correctly in the Jira comment created by the bot.

**Pull request merged**